bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -448,6 +448,7 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
                             ]
                             val oldThing = thingsMap.get(modelName).findFirst[it.UID == newThing.UID]
                             if (oldThing != null) {
+                                newThing.merge(oldThing)
```

Comment:
IMHO it is good to merge the old stuff to the new because we can keep the references of the old and new thing.
WDYT about adding an attribute "preserve" attribute to the merge function (default = false, so keep the a function  that calls `thing.merge(arg, false)`) and it only adds non-existing stuff from the old to the new thing (e.g. channels that exist in the old one but not in the new one), but keep the existing information (e.g. keep bridge id, location of the existing (here newThing))?